src/Gui/Forms/InitialPageInteractor.cs

```diff
@@ -182,5 +192,78 @@ private bool NeedsScanning()
                 Decompiler.Project != null &&
                 Decompiler.Project.Programs.Any(p => p.NeedsScanning);
         }
+
+        private void ProcessResources(ProgramResourceGroup prg, string outputDir)
+        {
+            try
+            {
+                if (prg.Name == "PE resources")
+                {
+                    // Create the dir
+                    Directory.CreateDirectory(outputDir + "\\resources");
+                    foreach (ProgramResourceGroup pr in prg.Resources)
+                    {
+                        switch (pr.Name)
+                        {
+                            case "CURSOR":
+                            {
+                                WriteResourceFile(outputDir + "\\resources", "Cursor", ".cur", pr);
+                            }
+                            break;
+                            case "BITMAP":
+                            {
+                                WriteResourceFile(outputDir + "\\resources", "Bitmap", ".bmp", pr);
+                            }
+                            break;
+                            case "ICON":
+                            {
+                                WriteResourceFile(outputDir + "\\resources", "Icon", ".ico", pr);
+                            }
+                            break;
+                            case "FONT":
+                            {
+                                WriteResourceFile(outputDir + "\\resources", "Font", ".bin", pr);
+                            }
+                            break;
+                            case "NEWBITMAP":
+                            {
+                                WriteResourceFile(outputDir + "\\resources", "NewBitmap", ".bmp", pr);
+                            }
+                            break;
+
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+
+            }
+        }
+
+        private void WriteResourceFile(string outputDir, string ResourceType, string ext, ProgramResourceGroup pr)
+        {
+            Directory.CreateDirectory(outputDir + "\\" + ResourceType);
+            foreach (ProgramResourceGroup pr1 in pr.Resources)
+            {
+                foreach (ProgramResourceInstance pr2 in pr1.Resources)
+                {
+                    // write the bytes to file
+                    try
+                    {
+                        using (var fs = new FileStream(outputDir + "\\" + ResourceType + "\\" + pr1.Name + ext, FileMode.Create, FileAccess.Write))
+                        {
+                            fs.Write(pr2.Bytes, 0, pr2.Bytes.Length);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+
```

Comment:
Again, the exception is being dropped on the floor. Since this method is being called in a loop, a better reporting mechanism is to use the `IDiagnosticsService.Error` method:
```
catch (Exception ex) {
    var diagSvc = Services.RequireService<IDiagnosticsService>();
    diagSvc.Error(ex, "An error occurred while displaying the program.");
}
```
Consider returning a `bool` from this method: return `true` if write was successful, and `false` if it failed. The calling method should `and` together all returned values from this method, and if the total result is `false`, show a dialog box with the message "One or more resource files was not saved successfully."